mm2src/coins/nft/trading_proto_v2/mod.rs

```diff
@@ -0,0 +1,565 @@
+use crate::coin_errors::{ValidatePaymentError, ValidatePaymentResult};
+use ethabi::{Contract, Token};
+use ethcore_transaction::{Action, UnverifiedTransaction};
+use ethereum_types::{Address, U256};
+use futures::compat::Future01CompatExt;
+use mm2_err_handle::prelude::{MapToMmResult, MmError, MmResult};
+use mm2_number::BigDecimal;
+use std::convert::TryInto;
+use web3::types::{Transaction as Web3Tx, TransactionId};
+
+pub(crate) mod errors;
+use errors::{Erc721FunctionError, HtlcParamsError, PaymentStatusErr};
+pub(crate) mod structs;
+use structs::{ExpectedHtlcParams, StateType, ValidationParams};
+
+use super::ContractType;
+use crate::eth::{addr_from_raw_pubkey, decode_contract_call, EthCoin, EthCoinType, MakerPaymentStateV2, SignedEthTx,
+                 TryToAddress, ERC1155_CONTRACT, ERC721_CONTRACT, ETH_GAS, NFT_SWAP_CONTRACT};
+use crate::nft::trading_proto_v2::errors::PrepareTxDataError;
+use crate::{NftAssocTypes, RefundPaymentArgs, SendNftMakerPaymentArgs, SpendNftMakerPaymentArgs, TransactionErr,
+            ValidateNftMakerPaymentArgs};
+
+impl EthCoin {
+    pub(crate) async fn send_nft_maker_payment_v2_impl(
+        &self,
+        args: SendNftMakerPaymentArgs<'_, Self>,
+    ) -> Result<SignedEthTx, TransactionErr> {
+        let contract_type = try_tx_s!(self.parse_contract_type(args.nft_swap_info.contract_type));
+        try_tx_s!(self.validate_payment_args(
+            args.taker_secret_hash,
+            args.maker_secret_hash,
+            &args.amount,
+            &contract_type,
+        ));
+
+        let taker_address = try_tx_s!(addr_from_raw_pubkey(args.taker_pub));
+        let token_address = try_tx_s!(self.parse_contract_address(args.nft_swap_info.token_address));
+        let swap_contract_address = try_tx_s!(self.parse_contract_address(args.nft_swap_info.swap_contract_address));
+        let time_lock_u32 = try_tx_s!(args.time_lock.try_into());
+        let token_id_u256 = U256::from(args.nft_swap_info.token_id);
+        let htlc_data = self.prepare_htlc_data(&args, taker_address, token_address, time_lock_u32);
+
+        match &self.coin_type {
+            EthCoinType::Nft { .. } => {
+                let data = try_tx_s!(self.prepare_nft_maker_payment_v2_data(
+                    contract_type,
+                    swap_contract_address,
+                    token_id_u256,
+                    &args,
+                    htlc_data
+                ));
+                self.sign_and_send_transaction(0.into(), Action::Call(token_address), data, U256::from(ETH_GAS))
```

Comment:
I guess this and other sign and send usages need to be modified here or in https://github.com/KomodoPlatform/komodo-defi-framework/pull/2051 depending on which PR gets merged first.
c.c. @dimxy